src/primitive_crypto.cc

```diff
@@ -210,25 +172,67 @@ PRIMITIVE(aes_init) {
   ByteArray* proxy = process->object_heap()->allocate_proxy();
```

Comment:
This needs to be moved up above the _new of the context, because otherwise we leak memory if the _new succeeds but the heap allocation of the the proxy fails.  The same issue doesn't arise in the opposite direction because the proxy is on the garbage collected heap so it gets collected at the next GC.

An added benefit is that you can combine the two 'if's and remove the cast.

Reply:
Also a couple of places with trailing spaces at the ends of lines.

Reply:
Done.